Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -194,103 +194,107 @@ public class ProgramBuilder {
     public func randVar(ofType type: Type) -> Variable? {
         var wantedType = type
-        
+
         if mode == .aggressive {
             wantedType |= .unknown
         }
-        
+
         if let v = randVarInternal({ self.type(of: $0).Is(wantedType) }) {
             return v
         }
-        
+
         // Didn't find a matching variable. If we are in aggressive mode, we now simply return a random variable.
         if mode == .aggressive {
             return randVar()
         }
-        
+
         // Otherwise, we give up
         return nil
     }
-    
+
     /// Returns a random variable of the given type. This is the same as calling randVar in conservative building mode.
     public func randVar(ofConservativeType type: Type) -> Variable? {
         let oldMode = mode
         mode = .conservative
         defer { mode = oldMode }
         return randVar(ofType: type)
     }
-    
+
     /// Returns a random variable satisfying the given constraints or nil if none is found.
     private func randVarInternal(_ selector: ((Variable) -> Bool)? = nil) -> Variable? {
         var candidates = [Variable]()
-        
+
         // Prefer inner scopes
         withProbability(0.75) {
             candidates = chooseBiased(from: scopeAnalyzer.scopes, factor: 1.25)
             if let sel = selector {
                 candidates = candidates.filter(sel)
             }
         }
-        
+
         if candidates.isEmpty {
             if let sel = selector {
                 candidates = scopeAnalyzer.visibleVariables.filter(sel)
             } else {
                 candidates = scopeAnalyzer.visibleVariables
             }
         }
-        
+
         if candidates.isEmpty {
             return nil
         }
-        
+
         return chooseUniform(from: candidates)
     }
-    
-    
+
+
     /// Type information access.
     public func type(of v: Variable) -> Type {
         return interpreter?.type(of: v) ?? .unknown
     }
-    
+
     public func methodSignature(of methodName: String, on object: Variable) -> FunctionSignature {
         return interpreter?.inferMethodSignature(of: methodName, on: object) ?? FunctionSignature.forUnknownFunction
     }
-    
+
     public func setType(ofProperty propertyName: String, to propertyType: Type) {
         interpreter?.setType(ofProperty: propertyName, to: propertyType)
     }
-    
+
+    public func setType(ofVariable variable: Variable, to variableType: Type) {
+        interpreter?.setType(of: variable, to: variableType)
+    }
+
     public func setSignature(ofMethod methodName: String, to methodSignature: FunctionSignature) {
         interpreter?.setSignature(ofMethod: methodName, to: methodSignature)
     }
-    
-    public func generateCallArguments(for signature: FunctionSignature) -> [Variable]? {
+
+    private func buildCallArgumentsInternal(for signature: FunctionSignature, block: (_: Type, _: inout [Variable]) -> Bool) -> [Variable]? {
```

Comment:
Hm, maybe we can make this `argumentTypes(for signature: FunctionSignature) -> [Type]`, and just let it produce an array of the types of the arguments, then call it from both `generateCallArguments` and `randCallArguments` and let them actually get/generate the argument values. I think that should simplify things a bit. WDYT?